### PR TITLE
test_engine_image_not_fully_deployed_perform_dr_restoring_expanding_volume

### PR DESCRIPTION
**Pull request by chriscchien**
Add test case `test_engine_image_not_fully_deployed_perform_dr_restoring_expanding_volume`

ref: [5974](https://github.com/longhorn/longhorn/issues/5974)

### CODE REVIEW

File: manager/integration/tests/test_ha.py [hunk 1]
```diff
@@ -3060,7 +3059,134 @@ def test_engine_image_not_fully_deployed_perform_dr_restoring_expanding_volume()
...
+    wait_for_backup_restore_completed(client, dr_volume_name, b1.name)
+    dr_volume = client.by_id_volume(dr_volume_name)
+
+    # step 4
+    on_node2 = False
+    on_node3 = False
+    for replica in dr_volume.replicas:
+
+        print()
+        if replica.hostId == node2.id:
```
Comment (innobead) [inline manager/integration/tests/test_ha.py:3109]: Print this dummy print.

Also plz remove unnecessary empty line. Usually we only use empty lines to group the related code for better understanding.

**Review by innobead — CHANGES_REQUESTED** [review]
In general, LGTM.

Some comments need to resolve, and also ensure the test cass run well.

File: manager/integration/tests/test_ha.py (hunk 1 above)
Comment (chriscchien) [reply manager/integration/tests/test_ha.py]: Sorry for the wrong code format, I will pay more attention in the future

**Comment by chriscchien** [comment]
Updated code and here is the test result on [pipeline](https://ci.longhorn.io/job/private/job/longhorn-tests-regression/3917/testReport/tests/test_ha/), thank you

**Review by innobead — APPROVED** [review]

**Comment by innobead** [comment]
@mergify backport v1.4.x v1.3.x

**Pull request merged**